_posts/2024-01-12-kasten-openshift-monitoring.markdown

```diff
@@ -0,0 +1,815 @@
+
+From an alerting point of view, what you need to do at the end is to capture when this expression will be true 
+
+```
+increase(catalog_actions_count{status='failed'}[10m])>0
+```
+
+This expression is a binary selector selector, if it evaluates to true it will return the instant vector that satifie the condition `> 0`.
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {
+          "container": "catalog-svc",
+          "endpoint": "http",
+          "exported_namespace": "mcourcy-pacman",
+          "instance": "10.131.1.77:8000",
+          "job": "catalog-svc",
+          "liveness": "live",
+          "namespace": "kasten-io",
+          "pod": "catalog-svc-56b97b4bfb-7vrzc",
+          "policy": "mcourcy-pacman-backup",
+          "prometheus": "openshift-user-workload-monitoring/user-workload",
+          "service": "catalog-svc",
+          "status": "failed",
+          "type": "backup"
+        },
+        "value": [
+          1705356529.062,
+          "1.0526315789473684"
+        ]
+      }
+    ]
+  }
+}   
+```
+
+In other words the exact result we get before, otherwise it just return an empty set
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": []
+  }
+}
+```
+
+Getting an empty resultset means that the number of failed action did not increase in the last 10 minutes.
+
+## Garbage collection of the metrics 
+
+Even long after the backup failure if you execute the request 
+```
+"query=increase(catalog_actions_count{policy='mcourcy-pacman-backup',status='failed'}[10m])"
+```
+
+You'll notice the date are still recent and within the last 10 minutes despite the failure is now over. The catalog continue to publish the metrics or the actions, and prometheus continue to scrape them. 
+
+This is because backup actions has not been removed and will be removed after the restorepoint will be removed.
+
+# Capturing other interesting failure 
+
+You can use the Kasten grafana dashboard to obtain other interesting alert 
+
+## Not using more than 50 % of catalog pvc 
+
+```
+100-catalog_persistent_volume_free_space_percent>50
+```
+
+If the result is empty you are under 50%.
+
+## Evaluating the quantity of memory and CPU used by kasten-io 
+
+```
+sum(container_memory_working_set_bytes{namespace='kasten-io',container="",pod!=""}) BY (namespace)
+```
+
+or 
+```
+curl -X GET -kG "https://$THANOS_QUERIER_HOST/api/v1/query?" --data-urlencode "query=sum(container_memory_working_set_bytes{namespace='kasten-io',container='',pod!=''}) BY (namespace)" -H "Authorization: Bearer $TOKEN" |jq
+```
+
+the result here 
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {
+          "namespace": "kasten-io"
+        },
+        "value": [
+          1705359310.344,
+          "2776227840"
+        ]
+      }
+    ]
+  }
+}
+```
+
+Show that kasten consumme 2.8Gb of memory.
```

Comment:
Is there any easy way to format the value (2776227840) into a human readable format? No worries if not, but might be worth including

Reply:
I don't think so with prometheus, it's why I provided the `date -d ` command